src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -353,15 +353,14 @@ static void CALLBACK executeOnRunLoopCallback(_In_ ULONG_PTR info) {
     } catch (const exception&) {
         command->failure = current_exception();
     }
-    unique_lock<mutex> lock(command->server->executionMutex);
+    unique_lock<mutex> lock(command->executionMutex);
```

Comment:
Looks like I don't understand the locking in C++. Why do we need this lock at all?

Reply:
It's not very intuitive I agree. This is required so that `wait_for()` always get called before `notify_all()`. Without this locking here you'd run into the reverse order, and `wait_for()` would hang.